src/ottplan.java

```diff
@@ -0,0 +1,141 @@
+import java.text.DecimalFormat;
+import java.util.*;
+abstract class Subscription{
+    protected String resolution;
+    protected String planName;
+    protected int noOfViewers;
+    protected double amount;
+    protected String dis;
+    abstract void calculate(boolean a);
```

Comment:
```suggestion
    protected int maxConcurrentViewers;
    protected double subscriptionFee;
    protected double taxRate;
    abstract void calculateTotalPrice()
```

Name the variables as intutively as possible. The variable name should be able to tell you what it stores.